app/src/main/AndroidManifest.xml

```diff
@@ -3,10 +3,8 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.hiddenramblings.tagmo">
 
-    <uses-permission android:name="android.permission.BIND_NFC_SERVICE"/>
     <uses-permission android:name="android.permission.NFC"/>
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>

```

Comment:
Will removing WRITE_EXTERNAL_STORAGE here prevent the app from saving scanned bin files to external storage?

Reply:
Thanks for pointing this out; after double checking, it turns out that the removal was based on a misunderstanding of the documentation:

https://developer.android.com/reference/android/Manifest.permission#WRITE_EXTERNAL_STORAGE
https://developer.android.com/reference/android/os/Environment#getExternalStorageDirectory()

This API will break from Android Q onward and the app needs an update to handle future devices.  But the devices with API level 23 and above will break if this is removed, so it should be kept.